File: python/arrays-and-hashing/remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        safe, last = 0, len(nums) - 1
        for i in range(len(nums)):
            if nums[i] != val:
                safe += 1
                continue

            while nums[last] == val:
                last -= 1

                if last <= i:
                    return safe

            nums[i], nums[last] = nums[last], nums[i]
            safe += 1

        return len(nums)

File: c++/arrays-and-hashing/remove-element.cc
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int last = nums.size() - 1;
        
        for (int i = 0; i <= last; ++i) {
            while (nums[last] == val && last > i) {
                last--;
            }

            if (nums[i] == val) {
                int temp = nums[i];
                nums[i] = nums[last];
                nums[last] = temp;
                last--;
            }
        }

        return last + 1;
    }
};
